File: Java/FactoryPattern/DatabaseFactory/src/com/laxmena/factory/SQLFactory.java
package com.laxmena.factory;

import com.laxmena.db.Database;
import com.laxmena.db.MySQLDatabase;
import com.laxmena.db.PostgresDatabase;

public class SQLFactory extends DatabaseFactory {

    @Override
    public Database createDatabase(String database) {
        if(database.equalsIgnoreCase("MySQL")) {
            return new MySQLDatabase();
        } else if(database.equalsIgnoreCase("Postgres")) {
            return new PostgresDatabase();
        } else {
            System.out.println("Requested database " + database + "is not supported yet.");
            return null;
        }
    }
}

File: FactoryPattern/Python/DatabaseAdapter/DatabaseAdapter/factory/sql_factory.py
from db.database import Database
from db.mysql_database import MySQLDatabase
from db.postgres_database import PostgresDatabase
from factory.database_factory import DatabaseFactory


class SQLFactory(DatabaseFactory):
    @staticmethod
    def create_database(name: str) -> Database:
        if name.lower() == "postgres":
            return PostgresDatabase()
        elif name.lower() == "mysql":
            return MySQLDatabase()
        else:
            print("Requested Database not found")
            return None
